src/api/gnosisProtocol/hooks.ts

```diff
@@ -117,3 +120,34 @@ export function useHasOrders(account?: string | null): boolean | undefined {
 
   return (gpOrders?.length || 0) > 0
 }
+
+export type UseSurplusAmountResult = {
+  surplusAmount: Nullish<CurrencyAmount<Currency>>
+  isLoading: boolean
+  error: string
+}
+
+export function useSurplusAmount(): UseSurplusAmountResult {
+  const { chainId, account } = useWalletInfo()
+  const nativeCurrency = useNativeCurrency()
+
+  const {
+    data: surplusAmount,
+    isLoading,
+    error,
+  } = useSWR<CurrencyAmount<Currency> | null>(['totalSurplus', chainId, account], async () => {
```

Comment:
what is the caching logic? 
Also, do you invalidate once the user has traded and got some?

Reply:
Good question, have not thought about that.
Will work on it in a follow up PR

Reply:
Thanks!

Reply:
linking https://github.com/cowprotocol/cowswap/pull/2781